synapse/events/snapshot.py

```diff
@@ -131,8 +119,9 @@ class EventContext(UnpersistedEventContextBase):
     _state_group: Optional[int] = None
     state_group_before_event: Optional[int] = None
     _state_delta_due_to_event: Optional[StateMap[str]] = None
-    prev_group: Optional[int] = None
-    delta_ids: Optional[StateMap[str]] = None
+    state_group_deltas: Optional[
+        Dict[Tuple[Optional[int], Optional[int]], Optional[StateMap[str]]]
```

Comment:
I'm a bit surprised these are all optional, those cases don't seem covered in the docstring!